frontend/src/app/exam/exam.component.ts

```diff
@@ -1,16 +1,255 @@
-import {Component, OnInit} from '@angular/core';
+import {AfterViewInit, ChangeDetectorRef, Component, OnInit, ViewEncapsulation} from '@angular/core';
+import {ExamService} from './service/exam.service';
+import {ExamDto} from './model/exam.dto';
+import {NotificationWrapperService} from '../core/notifications/notification-wrapper.service';
+import {ExamTaskDto} from './model/exam.task.dto';
+import {FormBuilder, FormControl, FormGroup} from '@angular/forms';
+import {ExamCreateUpdateDialog} from './create-update-dialog/exam-create-update-dialog.component';
+import {MdDialog} from '@angular/material';
+import {ResponsiveHelperService} from '../core/services/ui/responsive-helper.service';
+import {MEDIUM_DIALOG} from '../core/util/const';
+import {ConfirmDialogService} from '../core/services/confirm-dialog.service';
+import {and, isNotEmpty, isTruthy} from '../core/util/helper';
+import {Util} from '../core/util/util';
+import * as objectAssign from 'object-assign';
+import {SubjectFilterDto} from '../task/model/subject.filter.dto';
+import {ActivatedRoute, Router} from '@angular/router';
+import {DateUtil} from '../core/services/date-util.service';
+import {OCMediaChange} from '../core/services/ui/oc-media-change';
+import {Observable} from 'rxjs/Observable';
+import {examByNameComparator} from '../core/util/comparator';
+import {Subject} from 'rxjs/Subject';
+import {MarkService} from 'app/mark/service/mark.service';
+import {ViewMode} from '../core/common/view-mode';
+import {slideUpDownAnimation} from '../core/animations/animations';
 
 @Component({
-    selector: 'app-exam',
+    selector: 'exam',
     templateUrl: './exam.component.html',
-    styleUrls: ['./exam.component.scss']
+    styleUrls: ['./exam.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    animations: [slideUpDownAnimation]
 })
-export class ExamComponent implements OnInit {
+export class ExamComponent implements OnInit, AfterViewInit {
 
-    constructor() {
+
+    private _activeExams: ExamDto[] = [];
+    private _allExams: ExamDto[] = [];
+    private _displayedExams: ExamDto[] = [];
+
+    private _searchForm: FormGroup;
+    private _filterForm: FormGroup;
+    private _filterData: SubjectFilterDto;
+    private _filterShown: boolean;
+    private _today: Date = new Date();
+
+    public addMark$: Subject<ExamDto> = new Subject();
+
+    constructor(private _examService: ExamService,
+                private _markService: MarkService,
+                private _router: Router,
+                private _route: ActivatedRoute,
```

Comment:
|`line 66`|
|---|
Don't understand how this works, it is never referenced and for me does not look like it has a source of items?